specs/lightclient/validator.md

```diff
@@ -0,0 +1,383 @@
+
+    contribution.signature = bls.Aggregate(signatures)
+    return contribution
+```
+
+##### Broadcast sync committee contribution
+
+If the validator is selected to aggregate (`is_sync_committee_aggregator`), then they broadcast their best aggregate as a `SignedContributionAndProof` to the global aggregate channel (`sync_committee_contribution_and_proof` topic) two-thirds of the way through the `slot`-that is, `SECONDS_PER_SLOT * 2 / 3` seconds after the start of `slot`.
+
+Selection proofs are provided in `ContributionAndProof` to prove to the gossip channel that the validator has been selected as an aggregator.
+
+`ContributionAndProof` messages are signed by the aggregator and broadcast inside of `SignedContributionAndProof` objects to prevent a class of DoS attacks and message forgeries.
+
+First, `contribution_and_proof = get_contribution_and_proof(state, validator_index, contribution, privkey)` is constructed.
+
+```python
+def get_contribution_and_proof(state: BeaconState,
+                               aggregator_index: ValidatorIndex,
+                               contribution: SyncCommitteeContribution,
+                               privkey: int) -> ContributionAndProof:
+    selection_proof = get_sync_committee_slot_signature(
+      state, 
+      contribution.slot,
+      privkey
+    )
+    return ContributionAndProof(
+        aggregator_index=aggregator_index,
+        contribution=contribution,
+        selection_proof=selection_proof,
+    )
+```
+
+Then `signed_contribution_and_proof = SignedContributionAndProof(message=contribution_and_proof, signature=signature)` is constructed and broadast. Where `signature` is obtained from:
+
+```python
+def get_contribution_and_proof_signature(
+  state: BeaconState,
+  contribution_and_proof: ContributionAndProof,
+  privkey: int) -> BLSSignature:
+    contribution = contribution_and_proof.contribution
+    domain = get_domain(state, DOMAIN_CONTRIBUTION_AND_PROOF, compute_epoch_at_slot(contribution.slot))
+    signing_root = compute_signing_root(contribution_and_proof, domain)
+    return bls.Sign(privkey, signing_root)
+```
+
+## Sync committee subnet stability
+
+The sync committee subnets need special care to ensure stability given the relatively low number of validators involved in the sync committee at any particular time. To provide this stability, a validator must do the following:
+
+* Maintain advertisement of the subnet the validator in the sync committee is assigned to in their node's ENR as soon as they have joined the subnet. Subnet assignments are known `EPOCHS_PER_SYNC_COMMITTEE_PERIOD` epochs in advance and can be computed with `compute_subnet_for_sync_committee` defined above. ENR advertisement is indicated by setting the `syncnets` entry in the ENR to a `uint8` encoding of the `subnet_id`, e.g. `ENR["syncnets"] = b'\x07'`. This entry is removed from the ENR after the validator has left the sync committee.
+
+- TODO: what happens for the first sync committee?
```

Comment:
may need to advertise as soon as you know which subnet you are in (so 1 sync committee period of lookahead) in order to give validating nodes time to find peers... seems like you can still join just ahead of the committee change but i'd have to revisit gossipsub to make sure this is feasible

Reply:
hmm, probably want to join ~1 to 5 epochs head of time and advertise when you join.